From 475e4a118d8a378f79abaf13fe996663fb59e420 Mon Sep 17 00:00:00 2001
From: Varadarajan Narayanan <quic_varada@quicinc.com>
Date: Wed, 22 Jan 2025 08:07:49 +0530
Subject: [PATCH 17/20] PENDING: arm: lib: Add EDL reset support in U-Boot
 Enable EDL

Add support for reset to edl.
"reset edl" command resets system into EDL mode.

Signed-off-by: Varadarajan Narayanan <quic_varada@quicinc.com>
Upstream-Status: Pending
---
 arch/arm/lib/reset.c             | 8 ++++++++
 arch/arm/mach-snapdragon/board.c | 6 ++++++
 2 files changed, 14 insertions(+)

diff --git a/arch/arm/lib/reset.c b/arch/arm/lib/reset.c
index c9796a4435..99f2ff3215 100644
--- a/arch/arm/lib/reset.c
+++ b/arch/arm/lib/reset.c
@@ -30,6 +30,10 @@ __weak void reset_misc(void)
 {
 }
 
+__weak void reset_edl(void)
+{
+}
+
 int do_reset(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 {
 	puts ("resetting ...\n");
@@ -37,6 +41,10 @@ int do_reset(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 
 	disable_interrupts();
 
+	/* if "reset edl" is provided */
+	if ((argc > 1) && (strcmp(argv[1], "edl") == 0))
+		reset_edl();
+
 	reset_misc();
 	reset_cpu();
 
diff --git a/arch/arm/mach-snapdragon/board.c b/arch/arm/mach-snapdragon/board.c
index 75a880f093..394af747ed 100644
--- a/arch/arm/mach-snapdragon/board.c
+++ b/arch/arm/mach-snapdragon/board.c
@@ -185,6 +185,12 @@ void *board_fdt_blob_setup(int *err)
 	return (void *)gd->fdt_blob;
 }
 
+void reset_edl(void)
+{
+	/* first argument is reset level and second argument is cookie value */
+	psci_system_reset2(0, 1);
+}
+
 void reset_cpu(void)
 {
 	psci_system_reset();
-- 
2.34.1

